frontend/src/hooks/useClickOutside.js

```diff
@@ -0,0 +1,20 @@
+import { useEffect } from "react";
+
+function useClickOutside(ref, handler) {
+  useEffect(() => {
+    const listener = (event) => {
+      if (!ref.current || ref.current.contains(event.target)) {
+        return;
+      }
+      handler(event);
+    };
+    document.addEventListener("mousedown", listener);
+    document.addEventListener("touchstart", listener);
+    return () => {
+      document.removeEventListener("mousedown", listener);
+      document.removeEventListener("touchstart", listener);
```

Comment:
try not to use `addEventListener` in react. read handling event from here: https://reactjs.org/docs/handling-events.html

Reply:
Its fine I think, I have seen this pattern used in production, also removing listener on unmount can't be done that way, so I am doing it like this.